plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -647,32 +655,51 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
         else {
             schemaTableNames = ImmutableList.of(prefix.toSchemaTableName());
         }
-        return schemaTableNames.stream()
```

Comment:
You could convert this to a parallel stream with a custom `ForkJoinPool`. This will avoid eager creation of all the tasks.